### ISSUE TITLE
Rule: disallow identical tests

### ISSUE THREAD

**Issue by not-an-aardvark** [not-an-aardvark/eslint-plugin-eslint-plugin#5]
When a rule has a lot of tests, it's sometimes difficult to tell if any tests are duplicates. This rule would warn if any test cases have the same properties.

Incorrect code for this rule:

```js
/* eslint eslint-plugin/no-identical-tests: error */

ruleTester.run("foo", rule, {
  valid: [
    { code: "foo", options: ["bar"] },
    { code: "somethingElse", options: ["bar"] },
    { code: "foo", options: ["bar"] }
  ],
  invalid: []
});
```

To be honest, I'm not sure how useful this would be in practice -- two test cases might be basically equivalent as far as the rule's logic is concerned, but have slight differences such as whitespace, and `eslint-plugin-eslint-plugin` wouldn't be able to tell that the cases were equivalent.
